### ISSUE TITLE
Support Faraday 2.0, without which Rails 7.0 upgrade is blocked

### ISSUE THREAD

**Issue by v-kumar** [googleapis/ruby-cloud-env#28]
Over the last year, most of the ruby gem ecosystem that uses Faraday http client library has migrated to Faraday 2.0. Most new gems only support Faraday 2.0.

See https://github.com/googleapis/google-auth-library-ruby/issues/355 for example.

I looked at the source code. There's no hard dependency on Faraday 1.x and the gemspec can simply be revised to 
```
gem.add_dependency "faraday", ">= 0.17.3", "< 3.0"
```

https://github.com/googleapis/ruby-cloud-env/blob/e18f199813ebba816b1cb0ea19b4254c408f883d/google-cloud-env.gemspec#L23

**Comment by multiplegeorges** [comment]
@dazuma I'm also hitting this issue.

```
In Gemfile:
    faraday-follow_redirects was resolved to 0.2.0, which depends on
      faraday (< 3, >= 2)

    google-cloud-storage was resolved to 1.36.1, which depends on
      google-cloud-core (~> 1.6) was resolved to 1.6.0, which depends on
        google-cloud-env (~> 1.0) was resolved to 1.5.0, which depends on
          faraday (>= 0.17.3, < 2.0)
```

**Comment by v-kumar** [comment]
Hi @shivgautam / @NivedhaSenthil : Can you please release a new dot version with the faraday changes?
